### ISSUE TITLE
More robust sync from git

### ISSUE THREAD

**Issue by mscherer** [elekto-io/elekto#72]
TLDR:

The current system of using a git checkout with either clone or and syncing is IMHO not robust enough, and should be replaced by a sync function that create a temporary directory, do a git clone, and then use that clone for the sync. This would prevent the issue that I dubbed "the mysterious case of the knative 2022-TOC disparition".

So earlier this week, @jberkus asked us to investigate why the elekto instance we setup on Openshift had lost 1 election after the upgrade to 0.6. Upon further debugging yesterday, we found this was due to the ballot table being empty, and after ruling out data corruptions and the like, we stopped investigating. But we found this happened before the update, which puzzled us a lot. 

I restarted the investigation this evening and found I think the cause.

So ballots are not removed from the db, unless [a election is removed](https://github.com/elekto-io/elekto/blob/main/elekto/models/sql.py#L313). Upon looking the database backups we had, I noticed that indeed, the election of 2022 wasn't here, and so my hypothesis at that time was that the election removal triggered the ballots cleanup. However, it was unclear how the election could be removed in the first place. Upon checking the code, I found that the sync function do remove elections if they are not in the yaml file for the election, and that sync is triggered by [the webhook](https://github.com/elekto-io/elekto/blob/main/elekto/controllers/webhook.py#L34). 

But the yaml file should never be absent, because the webhook populate them the git repo if it doesn't exist, correct ? Yes, but... the sync function can also be run from the CLI using --sync, and this is done when [starting the container](https://github.com/elekto-io/elekto/blob/main/entrypoint.sh#L8)

I wasn't aware that the meta directory had to persist (because that's not how I would have done). So I placed it [on /tmp](https://gitlab.com/osci/ansible-role-openshift-apps/-/blob/main/elekto/tasks/main.yml#L77), which is likely wiped on restart, or cleaned somehow, for example, when we have a upgrade as it happen on a regular basis.

So the 1st obvious fix is to tell "make sure META_PATH is persistent". But I think a better fix would be to not need META_PATH in the first place, and have the sync() function do a git clone (maybe a shallow one) each time it is called, in a temporary directory and operate from here. 

Besides avoiding the problem I faced, this also would be more robust in case of container crash, for example, in the middle of a git clone that leave some lock file and requires a human to fix. It would also simplify the code a bit, by removing 1 option.

**Comment by jberkus** [comment]
So the issue here is failing to check for an exception correctly?  Will look into it.

Also, are you saying that you think that maybe there was a container misfire, where /tmp/ was not writeable, and sync deleted the elections for that reason?

**Comment by jberkus** [comment]
I don't think only 2022-TOC disappeared.  I think all three elections disappeared; it's that only 2022-TOC was not backed up.

**Comment by jberkus** [comment]
Addressing part of this with #76

**Comment by jberkus** [comment]
The other parts of this issue have been fixed.  Closing this and tracking the remaining enhancement via #74
